lib/fortnox/api/types/nullable.rb

```diff
@@ -8,13 +8,17 @@ module Types
       module Nullable
         String = Types::Strict::String.optional.constructor { |value| value&.to_s }
         Float = Types::Strict::Float.optional.constructor { |value| value&.to_f }
-        Integer = Types::Strict::Int.optional.constructor { |value| value&.to_i }
-        Boolean = Types::Bool.optional.constructor { |value| THE_TRUTH.fetch(value) { false } }
-        # TODO: Improve parsing!
-        # In case date parsing fails, ArgumentError is thrown. Currently, it is rescued in Repository::Loaders.find.
-        # That method assumes that the exception is due to invalid argument to the find method, which is not the case
-        # if it is raised from here!
-        Date = Types::Date.optional.constructor { |value| ::Date.parse(value.to_s) unless value.nil? || value == '' }
+        Integer = Types::Strict::Integer.optional.constructor { |value| value&.to_i }
+        Boolean = Types::Bool.optional.constructor { |value| THE_TRUTH.fetch(value, false) }
+        Date = Types::Date.optional.constructor do |value|
+          next if value.nil? || value == ''
```

Comment:
I don't think `next` is correct here. We probably want to do `return nil` instead, to avoid letting `''` through as a nil value. Or does that get handled internally by Dry::Types?

Reply:
We can't use `return` in a `constructor`... That's why we are using `next`.

Reply:
It's handled by `Dry::Types`. I added a test for it and it works as expected. `''` gives `nil`.